app/models/client_opportunity_match.rb

```diff
@@ -405,7 +405,14 @@ def self.send_summary_emails
     end.flatten.uniq
 
     Contact.where(id: contact_ids).find_each do |contact|
-      MatchDigestMailer.digest(contact).deliver_now
+      if contact.user.present?
```

Comment:
Let's just 
```suggestion
      return unless contact.user.present?
```
Every contact _should_ have a user, if they don't, there's no reason to send a summary, they won't be able to login.